bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -32,8 +33,18 @@ import Scalaz._
 import scalaz.effect.IO
 import scalaz.effect.IO.ioUnit
 
+/** The status of BAGS for a given observation, for UI feedback purposes. */
+sealed class BagsStatus
+
+object BagsStatus {
+  case object ErrorStatus                extends BagsStatus
+  case object PendingStatus              extends BagsStatus
+  case object RunningStatus              extends BagsStatus
+  case class  FailureStatus(why: String) extends BagsStatus
+}
+
```

Comment:
That was my suggestion, but maybe I steered @sraaphorst  wrong on this one.  I didn't want the state transition methods being called in the UI but thinking about it for a second there's no real reason to worry -- they don't _do_ anything other than return a new state and an unexecuted action.